package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -123,24 +147,64 @@ def _mda_topology_from_omm_topology(self, omm_topology):
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
 
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
-
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        validated_elements = []
+        masses = []
+        atomtypes = []
+        for a in omm_topology.atoms():
+            elem = a.element
+            if elem is not None:
+                if elem.symbol.capitalize() in SYMB2Z:
+                    validated_elements.append(elem.symbol)
+                else:
+                    validated_elements.append('')
```

Comment:
when is this branch reached?

Reply:
That branch is reached in case of invalid elements. We use empty string for elements but use the element symbol as is for atomtype . [Context](https://github.com/MDAnalysis/mdanalysis/pull/3511#discussion_r843332961)